src/frontend/src/views/Upload/components/Metadata/components/Table/components/Row/components/LocationField/index.tsx

```diff
@@ -1,75 +1,119 @@
-import { MenuItem } from "czifui";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import { DefaultMenuSelectOption, Dropdown } from "czifui";
 import { FormikContextType } from "formik";
-import React from "react";
-import { Metadata } from "src/views/Upload/components/common/types";
+import { escapeRegExp } from "lodash/fp";
+import React, { useReducer } from "react";
+import {
+  Metadata,
+  NamedGisaidLocation,
+} from "src/views/Upload/components/common/types";
 import ApplyToAllColumn from "../common/ApplyToAllColumn";
-import { COUNTIES } from "./COUNTIES";
-import { MenuSubtext, StyledTextField } from "./style";
-
-const SELECT_PROPS = {
-  displayEmpty: true,
-  renderValue,
-};
+import { StyledDiv } from "./style";
 
 interface Props {
   fieldKey: keyof Metadata;
   formik: FormikContextType<Metadata>;
   applyToAllColumn: (fieldKey: keyof Metadata, value: unknown) => void;
   isFirstRow: boolean;
+  locations: NamedGisaidLocation[];
+}
+
+interface LocationSearchState {
+  searching?: boolean;
+  results: NamedGisaidLocation[];
+}
+
+function locationSearchReducer(
+  state: LocationSearchState,
+  action: LocationSearchState | Partial<LocationSearchState>
+): LocationSearchState {
+  return { ...state, ...action };
 }
 
 export default function LocationField({
   fieldKey,
   formik,
   applyToAllColumn,
   isFirstRow,
+  locations,
 }: Props): JSX.Element {
-  const { handleChange, handleBlur, values, touched, errors } = formik;
+  const { handleBlur, setFieldValue, values, touched, errors } = formik;
 
   const errorMessage = touched[fieldKey] && errors[fieldKey];
 
-  const value = values[fieldKey] || "";
+  let value: NamedGisaidLocation | null = null;
+  if (values[fieldKey]) {
+    value = values[fieldKey] as NamedGisaidLocation;
+  }
 
-  return (
-    <StyledTextField
-      select
-      SelectProps={SELECT_PROPS}
-      name={fieldKey}
-      value={value}
-      margin="dense"
-      variant="outlined"
-      onChange={handleChange}
-      onBlur={handleBlur}
-      error={Boolean(errorMessage)}
-      helperText={
-        errorMessage ||
-        (isFirstRow && value && (
-          <ApplyToAllColumn
-            fieldKey={fieldKey}
-            value={value}
-            handleClick={applyToAllColumn}
-          />
-        ))
+  const [state, dispatch] = useReducer(locationSearchReducer, {
+    results: [],
+    searching: false,
+  });
+
+  const searcher = async (query: string): Promise<void> => {
```

Comment:
why did you decide to manage the search state here instead of relying on the built-in search behavior?